src/openrct2/world/map_generator/SimplexNoise.h

```diff
@@ -9,14 +9,61 @@
 
 #pragma once
 
+#include "Noise.h"
+
 #include <cstdint>
 
+#ifdef __WARN_SUGGEST_FINAL_METHODS__
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wsuggest-final-methods"
+    #pragma GCC diagnostic ignored "-Wsuggest-final-types"
+#endif
+
 namespace OpenRCT2::World::MapGenerator
 {
     struct Settings;
 
-    void NoiseRand();
-    float FractalNoise(int32_t x, int32_t y, float frequency, int32_t octaves, float lacunarity, float persistence);
+    class SimplexNoise : public Noise
+    {
+    private:
+        uint8_t _perm[512];
+
+    public:
+        SimplexNoise(uint32_t seed);
+        SimplexNoise();
+        float Generate(float x, float y) override;
+    };
+
+    class SimplexFbmNoise : public SimplexNoise
```

Comment:
Another unnecessary abstraction, I don't see anything instantiate SimplexNoise.

Reply:
See other comment, the wgen plugin branch exposes the non fbm noise through the api.
There are cases where you'd want to roll your own frequency summation instead of using fbm.